js/core/options/option_manager.js

```diff
@@ -0,0 +1,100 @@
+import createCallBack from '../utils/callbacks';
+import { compileGetter, compileSetter } from '../utils/data';
+import { equals } from '../utils/comparator';
+import { extend } from '../utils/extend';
+import { isDefined, isPlainObject } from '../utils/type';
+import { normalizeOptions } from './utils';
+
+export class OptionManager {
+    constructor(options, optionsByReference) {
+        this._options = options;
+        this._optionsByReference = optionsByReference;
+
+        this._changingCallbacks = createCallBack({ syncStrategy: true });
+        this._changedCallbacks = createCallBack({ syncStrategy: true });
+        this._namePreparedCallbacks = createCallBack({ syncStrategy: true });
+
+        this.cachedGetters = {};
+        this.cachedSetters = {};
+    }
+
+    _setByReference(options, rulesOptions) {
+        extend(true, options, rulesOptions);
+
+        for(const fieldName in this._optionsByReference) {
+            if(Object.prototype.hasOwnProperty.call(rulesOptions, fieldName)) {
+                options[fieldName] = rulesOptions[fieldName];
```

Comment:
Should we use `silent` method for these operations?

Reply:
`_setByReference` is called by `silent`. It will be infinite loop.